packages/gatsby/src/utils/flags.ts

```diff
@@ -235,16 +235,6 @@ const activeFlags: Array<IFlag> = [
     experimental: false,
     testFitness: (): fitnessEnum => true,
   },
-  {
-    name: `GRAPHQL_TYPEGEN`,
-    env: `GATSBY_GRAPHQL_TYPEGEN`,
-    command: `develop`,
-    telemetryId: `GraphQLTypegen`,
-    description: `More easily incorporate content into your pages through automatic TypeScript type generation and better GraphQL IntelliSense. See https://github.com/gatsbyjs/gatsby/discussions/35420 for more details.`,
-    experimental: false,
-    noCI: true,
-    testFitness: (): fitnessEnum => true,
-  },
```

Comment:
We should never remove flags (at least within same major version) as this will cause "unknown flag error" without contextual information that this is now handled via top level gatsby-config field.

I think instead we could keep flag and in `testFitness` function do:
 1. return `false` (so we just don't set env var as it's no longer used)
 2. provide warning message (would have to see what exactly happen when we do logging there)
 3. possibly try to set `config.graphqlTypegen` to true (if it wasn't set before), if we would want to prevent users having to "migrate" to using config immediately ( not strong feeling on this, maybe warning or actual panic() would be fine to force users to adjust config? )

Reply:
We have https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/handle-flags.ts#L193 but this won't work because

1) The `name` doesn't fit
2) We don't set it to `LOCKED_IN` but `false`

Reply:
Can we use `requires` field to show contextual information - https://github.com/gatsbyjs/gatsby/blob/09a6dd9e3c8ebbf9ce5ab40f70c18a1e8562424c/packages/gatsby/src/utils/handle-flags.ts#L111-L119 ?

Reply:
Oh yeah, that should do it. Good catch!

Reply:
done